src/main/java/appeng/helpers/PatternHelper.java

```diff
@@ -85,7 +85,13 @@ public PatternHelper( final ItemStack is, final World w )
 
 		for( int x = 0; x < inTag.tagCount(); x++ )
 		{
-			final ItemStack gs = new ItemStack( inTag.getCompoundTagAt( x ) );
+			NBTTagCompound ingredient = inTag.getCompoundTagAt( x );
+			final ItemStack gs = new ItemStack( ingredient );
+
+			if( !ingredient.hasNoTags() && gs.isEmpty() )
+			{
+				throw new IllegalArgumentException( "No pattern here!" );
```

Comment:
It is actually more about the order. `ingredient` is never used again, but only validated after it was used. One of these small things, which leaves you confused, when looking at it a few months later and you try to remember what special cases did need it to be in that strange order.

Which is pretty much the reason, why throwing the exception sounds strange and I try to remember, if there was something special about processing patterns, which needed it actually.